lnst/Controller/RunSummaryFormatters/JsonRunSummaryFormatter.py

```diff
@@ -137,6 +138,12 @@ def _transform_result(self, result: BaseResult) -> Optional[dict]:
                 "measurement_type": result.measurement_type,
                 "data": measurement_data,
             }
+        elif isinstance(result, Result):
+            return ret | {
+                "type": "arbitrary_data",
+                "description": result.description,
+                "data": result.data,
```

Comment:
this is potentially dangerous if the `Result.data` object isn't JSONSerializable... e.g. if it is some custom object

Reply:
So, I added this so that the PacketAssert results are formatted properly. I agree that this should be fixed, I'm now thinking that maybe this should be fixed in the https://github.com/LNST-project/lnst/blob/dda250f70b1953925bd579a8fb431ebcea692313/lnst/RecipeCommon/PacketAssert.py#L87 - here we should define custom result type that will be safely converted to serializable data.

Reply:
Followup in https://github.com/LNST-project/lnst/issues/414